### PR TITLE
Add http_proxy to client & Fix deviceflow

### PR DESCRIPTION
**Pull request by ByronHsu**
# TL;DR

1. Add http_proxy to oauth2 client because in on-prem cluster use cases, the client might not have access to the external network
2. Remove `verification_url_complete` because that field not exists in [azure ad](https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code#device-authorization-request) and the oauth2 standard.
3. Concat the scopes by white space per [the documentation](https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code#device-authorization-request)

## Type
 - [x] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3620

### CODE REVIEW

**Review by pingsutw — COMMENTED** [review]
could you run make fmt & lint locally?

File: flytekit/clients/auth/authenticator.py [hunk 1]
```diff
@@ -219,21 +222,20 @@ def __init__(
...
+            endpoint=endpoint, header_key=header_key or cfg.header_key, credentials=KeyringStore.retrieve(endpoint), http_proxy_url=http_proxy_url
         )
 
     def refresh_credentials(self):
-        resp = token_client.get_device_code(self._device_auth_endpoint, self._client_id, self._audience, self._scope)
+        resp = token_client.get_device_code(self._device_auth_endpoint, self._client_id, self._audience, self._scope, self._http_proxy_url)
         print(
             f"""
 To Authenticate navigate in a browser to the following URL: {resp.verification_uri} and enter code: {resp.user_code}
-OR copy paste the following URL: {resp.verification_uri_complete}
```
Comment (wild-endeavor) [inline flytekit/clients/auth/authenticator.py:230]: can we check for this field's existence, and show it if it's there?  Even if it's not part of the standard, it's nice to have.

File: flytekit/clients/auth/token_client.py [hunk 2]
```diff
@@ -31,15 +31,13 @@ class DeviceCodeResponse:
     {'device_code': 'code',
          'user_code': 'BNDJJFXL',
          'verification_uri': 'url',
-         'verification_uri_complete': 'url',
```
Comment (wild-endeavor) [inline flytekit/clients/auth/token_client.py:34]: this is okay to take out.

**Review by wild-endeavor — COMMENTED** [review]
two small changes otherwise +1

File: flytekit/clients/auth/authenticator.py (hunk 1 above)
Comment (ByronHsu) [reply flytekit/clients/auth/authenticator.py:230]: i don't think it should be there. resp.verification_uri + resp.user_code is enough for login.

File: flytekit/clients/auth/token_client.py (hunk 2 above)
Comment (ByronHsu) [reply flytekit/clients/auth/token_client.py:34]: wdym. to revert this?

**Review by wild-endeavor — DISMISSED** [review]
but needs lint

File: flytekit/clients/auth/authenticator.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/clients/auth/authenticator.py:230]: yeah but okta supports it.  and lots of people use okta.  it's not in the standard but i feel it's okay to support a nice to have.

File: flytekit/clients/auth/token_client.py (hunk 2 above)
Comment (wild-endeavor) [reply flytekit/clients/auth/token_client.py:34]: nah i mean it's okay to take it out of the test.  don't revert this.

File: flytekit/clients/auth/authenticator.py (hunk 1 above)
Comment (ByronHsu) [reply flytekit/clients/auth/authenticator.py:230]: @kumare3 wdyt ^^

Comment (wild-endeavor) [reply flytekit/clients/auth/authenticator.py:230]: can we just add back and merge?

**Review by pingsutw — APPROVED** [review]
